src/server/acl/validator.cc

```diff
@@ -4,12 +4,23 @@
 
 #include "server/acl/validator.h"
 
+#include "base/logging.h"
+#include "server/acl/acl_commands_def.h"
+#include "server/server_state.h"
+
 namespace dfly::acl {
 
 [[nodiscard]] bool IsUserAllowedToInvokeCommand(const ConnectionContext& cntx,
                                                 const facade::CommandId& id) {
-  auto command_credentials = id.acl_categories();
-  return (cntx.acl_categories & command_credentials) != 0;
+  auto cat_credentials = id.acl_categories();
+
+  const auto& commands = acl::CommandsIndexer();
+  DCHECK(commands.contains(id.name()));
+  auto [index, command_mask] = commands.find(id.name())->second;
+  DCHECK_LT(index, cntx.acl_commands.size());
```

Comment:
Actually, on a second thought 😅  we don't need any of the global stores if we store the family index & command index in the CommandId itself and as a bonus we save a hashmap lookup... I don't see any downsides in storing this info there

Reply:
I think this is a good idea, I will get back on this. I won't merge this PR. Let's make it as good as possible.